modules/xfem/src/userobjects/InterfaceMeshCut2DUserObject.C

```diff
@@ -0,0 +1,267 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "InterfaceMeshCut2DUserObject.h"
+#include "XFEMMovingInterfaceVelocityBase.h"
+
+registerMooseObject("XFEMApp", InterfaceMeshCut2DUserObject);
+
+InputParameters
+InterfaceMeshCut2DUserObject::validParams()
+{
+  InputParameters params = InterfaceMeshCutUserObjectBase::validParams();
+  params.addClassDescription(
+      "Creates a UserObject for a mesh cutter in 2D material interface problems");
+  return params;
+}
+
+InterfaceMeshCut2DUserObject::InterfaceMeshCut2DUserObject(const InputParameters & parameters)
+  : InterfaceMeshCutUserObjectBase(parameters)
+{
+  for (const auto & elem : _cutter_mesh->element_ptr_range())
+    if (elem->type() != EDGE2)
+      mooseError(
+          "InterfaceMeshCut2DUserObject currently only supports TRI3 element in the cut mesh.");
+}
+
+void
+InterfaceMeshCut2DUserObject::initialize()
+{
+  std::vector<Point> new_position(_cutter_mesh->n_nodes());
+  _pl = _mesh.getPointLocator();
+  _pl->enable_out_of_mesh_mode();
+
+  std::map<unsigned int, Real> node_velocity;
+  Real sum = 0.0;
+  unsigned count = 0;
+  for (const auto & node : _cutter_mesh->node_ptr_range())
+  {
+    if ((*_pl)(*node) != nullptr)
+    {
+      Real velocity;
+      if (_func == nullptr)
+        velocity =
+            _interface_velocity->computeMovingInterfaceVelocity(node->id(), nodeNomal(node->id()));
+      else
+        velocity = _func->value(_t, *node);
+
+      // only updates when t_step >0
+      if (_t_step <= 0)
+        velocity = 0.0;
+
+      node_velocity[node->id()] = velocity;
+      sum += velocity;
+      count++;
+    }
+  }
```

Comment:
Could you add just one or two sentences before for loops like this? I think that would be very helpful for reviewers and future developers.

Reply:
done.